### PR TITLE
drivers: avoid referencing client/structs package

### PR DESCRIPTION
**Pull request by notnoop**
Avoid drivers explicitly referencing the internal package, `nomad/client/structs`.

Moved `FSIsolation` and `DeviceNetwork` to drivers, but re-exported ResourceUsage interfaces to avoid drivers directly depending on the internal client/structs package directly.

I attempted moving the structs to drivers, but that caused some import cycles that was a bit hard to disentangle.  Alternatively, I added an alias here that's sufficient for our purposes of avoiding external drivers depend on internal packages, while allowing us to restructure packages in future without breaking source compatibility.

### CODE REVIEW

File: client/allocdir/task_dir.go [hunk 1]
```diff
@@ -142,15 +141,9 @@ func (t *TaskDir) Build(chrootCreated bool, chroot map[string]string, fsi cstruc
 // buildChroot takes a mapping of absolute directory or file paths on the host
 // to their intended, relative location within the task directory. This
 // attempts hardlink and then defaults to copying. If the path exists on the
-// host and can't be embedded an error is returned. If chrootCreated is true
-// skip expensive embedding operations and only ephemeral operations (eg
-// mounting /dev) are done.
-func (t *TaskDir) buildChroot(chrootCreated bool, entries map[string]string) error {
-	if !chrootCreated {
```
Comment (nickethier) [inline client/allocdir/task_dir.go]: Why get rid of this check?

Comment (nickethier) [reply client/allocdir/task_dir.go]: Hmm I guess because it's only called /w false.

Comment (notnoop) [reply client/allocdir/task_dir.go]: Indeed - `buildChroot` and `taskdir.Build()`[1] always is called with false, so didn't see the value.

[1] https://github.com/hashicorp/nomad/pull/5157/files/34ee0ba6b931f64dd5a97ca9df0c44506c17f07e#diff-301b205830d382e5f5ce9d97cd3e3404L78

Comment (nickethier) [reply client/allocdir/task_dir.go]: :+1:

File: client/allocdir/task_dir.go [hunk 2]
```diff
@@ -142,15 +141,9 @@ func (t *TaskDir) Build(chrootCreated bool, chroot map[string]string, fsi cstruc
...
-		// Link/copy chroot entries
-		return t.embedDirs(entries)
-	}
-	return nil
+// host and can't be embedded an error is returned.
+func (t *TaskDir) buildChroot(entries map[string]string) error {
```
Comment (schmichael) [inline client/allocdir/task_dir.go]: This func seems like a useless wrapper now.

Comment (notnoop) [reply client/allocdir/task_dir.go]: Indeed, I liked having it around, as the function name and comment is useful.  `embedDirs()` doesn't convey the same thing.

**Pull request merged**